rllib/core/models/specs/specs_base.py

```diff
@@ -87,26 +91,67 @@ class TensorSpec(Spec):
         validate: Checks if the shape and dtype of the tensor matches the
             specification.
         fill: creates a tensor with the specified value that is an
-            example of a tensor that matches the specification.
-
-    Abstract Methods:
-        get_type: Returns the type of the tensor, e.g. tf.Tensor or torch.Tensor.
-        get_shape: Returns the shape of the tensor depending on the backend.
-        get_dtype: Returns the dtype of the tensor depending on the backend.
-        _full: Creates a tensor with the specified value that
-            has values of fill_value, shape of shape, and dtype of self.dtype.
+            example of a tensor that matches the specification. This can only be
+            called if `framework` is specified.
     """
 
     def __init__(
         self,
         shape: str,
         *,
         dtype: Optional[Any] = None,
+        framework: Optional[str] = None,
         **shape_vals: int,
     ) -> None:
         self._expected_shape = self._parse_expected_shape(shape, shape_vals)
         self._full_shape = self._get_full_shape()
         self._dtype = dtype
+        self._framework = framework
+
+        # Import and prepare framework-specific properties.
+        if self._framework == "torch":
+            torch, _ = try_import_torch()
+            self._expected_type = torch.Tensor
+
+            def _full(cls, shape, fill_value=0):
+                return torch.full(shape, fill_value, dtype=cls.dtype)
+
+        elif self._framework == "tf2":
+            _, tf, _ = try_import_tf()
+            self._expected_type = tf.Tensor
+
+            def _full(cls, shape, fill_value=0):
```

Comment:
this doesn't need to be a inner function. Lets change that.

Reply:
Yep, thanks for catching it!

Reply:
maybe make it a non member function.